src/pyhf/optimize/common.py

```diff
@@ -0,0 +1,131 @@
+"""Common Backend Shim to prepare minimization for optimizer."""
+from .. import get_backend
+from ..tensor.common import _TensorViewer
+
+
+def _get_tensor_shim():
+    """
+    A shim-retriever to lazy-retrieve the necessary shims as needed.
+
+    Because pyhf.tensor is a lazy-retriever for the backends, we can be sure
+    that tensorlib is imported correctly.
+    """
+    tensorlib, _ = get_backend()
+    if tensorlib.name == 'numpy':
+        from .opt_numpy import wrap_objective as numpy_shim
+
+        return numpy_shim
+
+    if tensorlib.name == 'tensorflow':
+        from .opt_tflow import wrap_objective as tflow_shim
+
+        return tflow_shim
+
+    if tensorlib.name == 'pytorch':
+        from .opt_pytorch import wrap_objective as pytorch_shim
+
+        return pytorch_shim
+
+    if tensorlib.name == 'jax':
+        from .opt_jax import wrap_objective as jax_shim
+
+        return jax_shim
+    raise ValueError(f'No optimizer shim for {tensorlib.name}.')
+
+
+def shim(
+    objective,
+    data,
+    pdf,
+    init_pars,
+    par_bounds,
+    fixed_vals=None,
+    do_grad=False,
+    do_stitch=False,
+):
+    """
+    Prepare Minimization for Optimizer.
+
+    Args:
+        objective (`func`): objective function
+        data (`list`): observed data
+        pdf (`pyhf.pdf.Model`): model
```

Comment:
Want this hyperlinked

```suggestion
        pdf (~pyhf.pdf.Model): model
```

Reply:
Note to self: This happens a lot so this should have a search and replace applies to it